src/storage/mod.rs

```diff
@@ -54,23 +54,33 @@ impl StorageImpl {
 }
 
 /// Represents a storage engine.
-#[async_trait]
 pub trait Storage: Sync + Send + 'static {
+    /// the following two result future types to avoid `impl Future` return different types when
+    /// impl `Storage`.
+    type CreateTableResultFuture<'a>: Future<Output = StorageResult<()>> + Sync + Send + 'a
+    where
+        Self: 'a;
+    type DropTableResultFuture<'a>: Future<Output = StorageResult<()>> + Sync + Send + 'a
```

Comment:
```suggestion
    type DropTableResultFuture<'a>: Future<Output = StorageResult<()>> + Send + 'a
```

Doesn't need to be `Sync`.

Reply:
updated